backend/app/evaluation/toxicity/run.py

```
@@ -0,0 +1,98 @@
+    all_metrics = {}
+
+    for validator_name, build_fn in VALIDATORS.items():
+        print(f"  Running {validator_name} on {dataset_name}...")
+        validator = build_fn()
+
+        with Profiler() as p:
+            df[f"{validator_name}_result"] = (
+                df[text_col]
+                .astype(str)
+                .apply(
+                    lambda x: p.record(lambda t: validator.validate(t, metadata={}), x)
```

Ruff (0.15.9)

[warning] Function definition does not bind loop variable `validator`

(B023)